### PR TITLE
Remove unneeded checks in RWLock wrapper

### PR DESCRIPTION
**Pull request by anosach-luxoft**

### CODE REVIEW

**Comment by anosach-luxoft** [comment]
@malirod, see 6e00994

**Comment by anosach-luxoft** [comment]
@malirod, @pestOO, @nk0leg please review

File: src/components/utils/src/rwlock_win.cc [hunk 1]
```diff
@@ -97,7 +136,8 @@ bool RWLock::TryAcquireForWriting() {
   return true;
 }
 
-bool RWLock::Release() {
+bool sync_primitives::RWLock::Impl::Release() {
+  sync_primitives::AutoLock lock(status_lock_);
```
Comment (pestOO) [inline src/components/utils/src/rwlock_win.cc:140]: Lock inside of Lock? Is it really good idea?

Comment (anosach-luxoft) [inline src/components/utils/src/rwlock_win.cc:140]: @pestOO, we have no choice. We need state bool flag here.

**Pull request merged**